docs/datatypes/validated.md

```diff
@@ -589,8 +589,61 @@ validatedMonad.tuple2(Validated.invalidNec[String, Int]("oops"), Validated.inval
 ```
 
 This one short circuits! Therefore, if we were to define a `Monad` (or `FlatMap`) instance for `Validated` we would
-have to override `ap` to get the behavior we want. But then the behavior of `flatMap` would be inconsistent with
-that of `ap`, not good. Therefore, `Validated` has only an `Applicative` instance.
+have to override `ap` to get the behavior we want. 
+
+```scala mdoc:silent:nest
+import cats.Monad
+
+implicit def inconsistentValidatedMonad[E: Semigroup]: Monad[Validated[E, *]] =
+  new Monad[Validated[E, *]] {
+    def flatMap[A, B](fa: Validated[E, A])(f: A => Validated[E, B]): Validated[E, B] =
+      fa match {
+        case Valid(a)     => f(a)
+        case i@Invalid(_) => i
+      }
+
+    def pure[A](x: A): Validated[E, A] = Valid(x)
+
+    @annotation.tailrec
+    def tailRecM[A, B](a: A)(f: A => Validated[E, Either[A, B]]): Validated[E, B] =
+      f(a) match {
+        case Valid(Right(b)) => Valid(b)
+        case Valid(Left(a)) => tailRecM(a)(f)
+        case i@Invalid(_) => i
+      }
+
+    override def ap[A, B](f: Validated[E, A => B])(fa: Validated[E, A]): Validated[E, B] =
+      (fa, f) match {
+        case (Valid(a), Valid(fab)) => Valid(fab(a))
+        case (i@Invalid(_), Valid(_)) => i
+        case (Valid(_), i@Invalid(_)) => i
+        case (Invalid(e1), Invalid(e2)) => Invalid(Semigroup[E].combine(e1, e2))
+      }
+  }
+```
+
+But then the behavior of `flatMap` would be inconsistent with that of `ap`, and this will violate one of the [FlatMap laws](https://github.com/typelevel/cats/blob/main/laws/src/main/scala/cats/laws/FlatMapLaws.scala), `flatMapConsistentApply`:
+
+```scala
+def flatMapConsistentApply[A, B](fa: F[A], fab: F[A => B]): IsEq[F[B]] = 
+  fab.ap(fa) <-> fab.flatMap(f => fa.map(f))
```

Comment:
Unfortunately the syntax probably needs some explanation, it's not particularly obvious.

```suggestion
// the `<->` syntax means "is equivalent to"
def flatMapConsistentApply[A, B](fa: F[A], fab: F[A => B]): IsEq[F[B]] = 
  fab.ap(fa) <-> fab.flatMap(f => fa.map(f))
```

Reply:
I added a slightly longer explanation, take a look if looks good